Bug: E5012 incorrectly used for return type mismatch, causing wrong caret message

## Description

Error code E5012 is being used for "return type mismatch" errors, but E5012 is defined as "assignment value count mismatch". This causes the error caret to display the wrong message.

## Steps to Reproduce

1. Create a function with a typed return
2. Return a value of the wrong type
3. Observe the error output

**Error Output:**
```
error[E5012]: return type mismatch: expected [string], got array
  --> file.ez:69:46
   |
69 |     lines = clean_and_validate_assembly_lines(lines)
   |                                              ^ assignment value count mismatch
   |
```

Notice the caret shows "assignment value count mismatch" but the error message says "return type mismatch".

## Root Cause

In `pkg/interpreter/evaluator.go:3165`:

```go
return newErrorWithLocation("E5012", line, col,
    "return type mismatch: expected %s, got %s", expectedType, actualType)
```

But in `pkg/errors/codes.go:184`, E5012 is defined as:

```go
E5012 = ErrorCode{"E5012", "multi-assign-count-mismatch", "assignment value count mismatch"}
```

The error code's template message ("assignment value count mismatch") is shown at the caret, while the custom message ("return type mismatch") is shown in the header.

## Expected Behavior

Return type mismatch should have its own error code with an appropriate message, or the caret should show the actual error message.

## Suggested Fix

Create a new error code specifically for return type mismatch:

```go
// In codes.go
E5014 = ErrorCode{"E5014", "return-type-mismatch", "return type mismatch"}

// In evaluator.go:3165
return newErrorWithLocation("E5014", line, col,
    "return type mismatch: expected %s, got %s", expectedType, actualType)
```

## Related

This issue is related to #1103 (array literal type inference) which can trigger this error.

--- a/pkg/errors/codes.go
+++ b/pkg/errors/codes.go
@@ -193,6 +193,7 @@ var (
 	E5021 = ErrorCode{"E5021", "panic", "explicit panic called"}
 	E5022 = ErrorCode{"E5022", "assertion-failed", "assertion condition was false"}
 	E5023 = ErrorCode{"E5023", "postfix-requires-integer", "postfix operator needs integer operand"}
+	E5024 = ErrorCode{"E5024", "return-type-mismatch", "return type mismatch"}
 )
 
 // =============================================================================
@@ -464,7 +465,7 @@ var errorCodesByString = map[string]ErrorCode{
 	"E5006": E5006, "E5007": E5007, "E5008": E5008, "E5009": E5009, "E5010": E5010,
 	"E5011": E5011, "E5012": E5012, "E5013": E5013, "E5014": E5014, "E5015": E5015,
 	"E5016": E5016, "E5017": E5017, "E5018": E5018, "E5019": E5019, "E5020": E5020,
-	"E5021": E5021, "E5022": E5022, "E5023": E5023,
+	"E5021": E5021, "E5022": E5022, "E5023": E5023, "E5024": E5024,
 	// Import
 	"E6001": E6001, "E6002": E6002, "E6003": E6003, "E6004": E6004, "E6005": E6005,
 	"E6006": E6006, "E6007": E6007, "E6008": E6008, "E6009": E6009,

--- a/pkg/interpreter/evaluator.go
+++ b/pkg/interpreter/evaluator.go
@@ -3259,7 +3259,7 @@ func createTypeMismatchError(val Object, expectedType string, line, col int) *Er
 	}
 
 	// Generic type mismatch
-	return newErrorWithLocation("E5012", line, col,
+	return newErrorWithLocation("E5024", line, col,
 		"return type mismatch: expected %s, got %s", expectedType, actualType)
 }
 

